### PR TITLE
[ONNX] Export weight_norm

### PR DESCRIPTION
**Pull request by neginraoof**
Export _weight_norm
Caffe2 tests are inplace

Looks like there is a conflicting behavior in torch.nn.utils.weight_norm regarding None dim.
Where dim could be negative for backwards axes, but when dim = None, it's overwitten to -1
https://github.com/pytorch/pytorch/blob/0c48092b2270d56cdab327bd1ff0ca89f5b7d569/torch/nn/utils/weight_norm.py#L10

For now, this symbolic to matches the current behavior. But this might need to be changed in the torch module.

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -1120,6 +1120,15 @@ def forward(self, input):
         x = torch.randint(10, (4, 2, 3, 4), dtype=torch.int32)
         self.run_test(ViewModel(), x)
 
+    def test_weight_norm(self):
+        model = torch.nn.utils.weight_norm(torch.nn.Linear(5, 10), dim=0)
```
Comment (lara-hdr) [inline test/onnx/test_pytorch_onnx_onnxruntime.py]: dim=0 is the default.
Could you use another dimension?

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -2027,3 +2022,18 @@ def group_norm(g, input, num_groups, weight, bias, eps, cudnn_enabled):
...
+
+
+@parse_args('v', 'v', 'i')
+def _weight_norm(g, weight_v, weight_g, dim):
+    rank = weight_v.type().dim()
+    if rank:
+        # W = g * ((v) / ||v||)
+        # Compute norm_except_dim for l2 norm
+        axes = list(range(rank))
+        axes.remove(dim)
```
Comment (lara-hdr) [inline torch/onnx/symbolic_opset9.py]: We should support dim=None as well (computes a norm over the entire weight tensor).
Could you verify if axes.remove(dim) fails when dim=None, and add a test case with dim=None as well?

**Review by lara-hdr — CHANGES_REQUESTED** [review]

File: torch/onnx/symbolic_opset9.py (hunk 2 above)
Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py]: Thanks Lara for mentioning.
Looks like there is a conflicting behavior in torch.nn.utils.weight_norm regarding None dim.
Where dim could be negative for backwards axes, but when dim = None, it's overwitten to -1
https://github.com/pytorch/pytorch/blob/0c48092b2270d56cdab327bd1ff0ca89f5b7d569/torch/nn/utils/weight_norm.py#L10

**Review by lara-hdr — APPROVED** [review]
LGTM

File: torch/onnx/symbolic_opset9.py (hunk 2 above)
Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py]: @BowenBao Could you please take a look as well? Thanks.

File: torch/onnx/symbolic_opset9.py [hunk 3]
```diff
@@ -2039,3 +2034,25 @@ def group_norm(g, input, num_groups, weight, bias, eps, cudnn_enabled):
...
+        if dim:
+            if dim < -1:
+                dim += rank
+            if dim != -1:
+                axes.remove(dim)
+        norm_v = norm(g, weight_v, 2, axes, 1)
+        div = g.op("Div", weight_v, norm_v)
+        return g.op("Mul", div, weight_g)
+    else:
+        return g.op("ATen", v, g, dim_i=dim, operator_s="_weight_norm")
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py]: v,g ==> weight_v, weight_g

**Review by BowenBao — CHANGES_REQUESTED** [review]
one slight typo

**Review by BowenBao — APPROVED** [review]

**Review by houseroad — APPROVED** [review]

**Comment by neginraoof** [comment]
@houseroad I just rebased this PR. CI looks fine and it is approved. Could you please take a look?

**Comment by neginraoof** [comment]
@houseroad Can we merge this? Thanks!
